AxonDeepSeg/morphometrics/compute_morphometrics.py

```diff
@@ -1,197 +1,251 @@
+        for i in range(len(ind_centroid[0])):
+            # Note: The value "i" corresponds to the label number of im_axon_label
+            im_centroid[ind_centroid[0][i], ind_centroid[1][i]] = i
+
+        # markers = ndi.label(local_maxi)[0]
+        # Watershed segmentation of axonmyelin using distance map
+        im_axonmyelin_label = morphology.watershed(-distance, im_centroid, mask=im_axonmyelin)
+        # Measure properties of each axonmyelin object
+        axonmyelin_objects = measure.regionprops(im_axonmyelin_label)
+
+    # DEBUG
+    # from matplotlib import colors
+    # from matplotlib.pylab import *
+    # random_cmap = matplotlib.colors.ListedColormap(np.random.rand(256, 3))
+    # matshow(im_axon_label, fignum=1, cmap=random_cmap), show()
+    # import datetime
+    # savefig('fig_' + datetime.datetime.strftime(datetime.datetime.now(), '%Y%m%d%H%M%S%f') + '.png', format='png',
+    #         transparent=True, dpi=100)
+
+    # Loop across axon property and fill up dictionary with morphometrics of interest
+    for prop_axon in axon_objects:
+        # Centroid
+        y0, x0 = prop_axon.centroid
+        # Solidity
+        solidity = prop_axon.solidity
+        # Eccentricity
+        eccentricity = prop_axon.eccentricity
+        # Axon equivalent diameter in micrometers
+        axon_diam = prop_axon.equivalent_diameter * pixelsize
+        # Axon area in µm^2
+        axon_area = prop_axon.area * (pixelsize ** 2)
+        # Axon orientation angle
+        orientation = prop_axon.orientation
+        # Add metrics to list of dictionaries
+        stats = {'y0': y0,
+                 'x0': x0,
+                 'axon_diam': axon_diam,
+                 'axon_area': axon_area,
+                 'solidity': solidity,
+                 'eccentricity': eccentricity,
+                 'orientation': orientation}
+        # Deal with myelin
+        if im_myelin is not None:
+            # Find label of axonmyelin corresponding to axon centroid
+            label_axonmyelin = im_axonmyelin_label[int(y0), int(x0)]
+            # TODO: use logger
+            # print(label_axonmyelin)
+            # print('x, y = {}, {}'.format(x0, y0))
+            if label_axonmyelin:
+                # Get corresponding index from axonmyelin_objects list
+                ind_axonmyelin = \
+                    [axonmyelin_object.label for axonmyelin_object in axonmyelin_objects].index(label_axonmyelin)
+                myelin_diam = axonmyelin_objects[ind_axonmyelin].equivalent_diameter * pixelsize
+                myelin_area = axonmyelin_objects[ind_axonmyelin].area * (pixelsize ** 2)
+                stats['myelin_diam'] = myelin_diam
+                stats['myelin_area'] = myelin_area
+                stats['gratio'] = np.sqrt(axon_area / myelin_area)
+            else:
+                # TODO: use logger
+                print('WARNING: Myelin object not found for axon centroid [{},{}]'.format(y0, x0))
```

Comment:
I got this warning for the following image:

![ads_mask](https://user-images.githubusercontent.com/1421029/54078336-01dded00-42a1-11e9-9b30-4b1287e99338.png)

`WARNING: Myelin object not found for axon centroid [150.0,800.0]`

I'm believe that it's the image at the top left; will this (myelin) functionality not work for axons objects that are cut off by the boundary? If so, we should document this so that they can be manually removed/considered as outliers by the user.

Reply:
Side note: The myelin diameter & area were all exactly equal for the three other objects, further confirming a bug from the other comment above.

Reply:
It's actually the top right one that wasn't being labelled as an `axonmyelin` object, weird...:

![im_axonmyelin_label](https://user-images.githubusercontent.com/1421029/54080390-0d450e80-42c9-11e9-9211-442a7e258ff5.png)

Reply:
Ah ha, no, it actually does get labelled, but as the value 0. Found out by changing the value of the label `i` to `i+1` here:

https://github.com/neuropoly/axondeepseg/blob/b3577b26945d2edbddd31e2045bf9b9d953e5872/AxonDeepSeg/morphometrics/compute_morphometrics.py#L68-L70

See:

![im_axonmyelin_label](https://user-images.githubusercontent.com/1421029/54080422-dcb1a480-42c9-11e9-974f-2476e88f527c.png)

And since that object was being labelled as `0`, this line was being skipped for that object:

https://github.com/neuropoly/axondeepseg/blob/b3577b26945d2edbddd31e2045bf9b9d953e5872/AxonDeepSeg/morphometrics/compute_morphometrics.py#L116

Looks like my solution above (`im_centroid[ind_centroid[0][i], ind_centroid[1][i]] = i + 1 `) works for me; is it clean enough for you?

Results:

<img width="1138" alt="Capture d’écran 2019-03-10 à 00 20 21" src="https://user-images.githubusercontent.com/1421029/54080440-59dd1980-42ca-11e9-9ca9-e5bce90186ee.png">

Reply:
p.s., the g-ratios are almost dead-on the simulated values here (defined as 0.7 for the first 3, then 0.9, per the settings described in #179, where the default g-ratio is 0.7). 🎉